v1-2/learn/how-to-use-java-interoperability.md

```diff
@@ -0,0 +1,939 @@
+    class: "a.b.c.Builder",
+    paramTypes: [{class: "a.b.c.Person", dimensions:2}, "int"]
+} external;
+
+function builderWithStudentList(handle list, int index) returns handle = @java:Constructor {
+    class: "a.b.c.Builder",
+    paramTypes: [{class: "a.b.c.Student", dimensions:2}, "int"]
+} external;
+```
+
+### Call java methods
+You can use the `java:@Method` annotation to link Ballerina functions with Java static and instance methods. There is a small but important difference in calling Java static methods vs calling instance methods. 
+
+#### Static methods
+Let’s first look at how to call a static method. The class “java.util.UUID” has a static method with the signature `static UUID randomString()`. 
+
+```ballerina
+import ballerinax/java;
+import ballerina/io;
+
+function randomUUID() returns handle = @java:Method {
+   name: "randomUUID",
+   class: "java.util.UUID"
+} external;
+
+public function main() {
+   handle uuid = randomUUID();
+   io:println(uuid);
+}
+```
+
+The `name` field is optional here. If the Ballerina function name is the same as the Java method name, you don’t have to specify the `name` field.
+
+```ballerina
+function randomUUID() returns handle = @java:Method {
+   class: "java.util.UUID"
+} external;
+```
+
+#### Instance methods
+Now, let’s look at how to call Java instance methods using the same `ArrayDeque` class in the package `java.util`. It can be used as a stack with its `pop` and `push` instance methods with the following method signatures. 
+
+```java
+E pop();
+void push(E e);
+```
+
+Here are the corresponding Ballerina functions that are linked to these methods. 
+
+```ballerina
+function pop(handle arrayDequeObj) returns handle = @java:Method {
+   class: "java.util.ArrayDeque"
+} external;
+
+function push(handle arrayDequeObj, handle e) = @java:Method {
+   class: "java.util.ArrayDeque"
+} external; 
+```
+
+If you compare these functions with the Java method signatures, you would notice the additional parameter `handle arrayDequeObj` in Ballerina functions. Let’s look at a sample usage to understand the reason. 
+
+```ballerina
+public function main() {
+   // Create a new instance of ArrayDeque
+   handle arrayDequeObj = newArrayDeque();
+
+   // Convert a Ballerina string to a Java string 
+   string str = “Ballerina”
+   handle handleStr = java:fromString(str);
+
+   push(arrayDequeObj, handleStr);
+   handle e = pop(arrayDequeObj);
+}
+```
+
+As you can see, you need to first construct an instance of the `ArrayDeque` class. The variable `arrayDequeObj` refers to an `ArrayDeque` object. Then, you need to pass this variable to both `pop` and `push` functions because the corresponding Java methods are instance methods of the class `ArrayDeque`. Therefore, you need an instance of the `ArrayDeque` class in order to invoke its instance methods. You can think of the variable `arrayDequeObj` as the method receiver.
```

Comment:
```suggestion
As you can see, you need to first construct an instance of the `ArrayDeque` class. The `arrayDequeObj` variable refers to an `ArrayDeque` object. Then, you need to pass this variable to both the `pop` and `push` functions because the corresponding Java methods are instance methods of the`ArrayDeque` class. Therefore, you need an instance of the `ArrayDeque` class in order to invoke its instance methods. You can think of the `arrayDequeObj` variable as the method receiver.
```